File: MonsterHunterStories/Monster.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MonsterHunterStories
{
	internal class Monster
	{
		private readonly uint mAddress;

		public Monster(uint address)
		{
			mAddress = address;
		}

		public String Name
		{
			get => SaveData.Instance().ReadText(mAddress, 30);
			set => SaveData.Instance().WriteText(mAddress, 30, value);
		}

		public uint ID
		{
			get => SaveData.Instance().ReadNumber(mAddress + 60, 2);
			set => SaveData.Instance().WriteNumber(mAddress + 60, 2, value);
		}

		public uint Lv
		{
			get => SaveData.Instance().ReadNumber(mAddress + 104, 1);
			set => Util.WriteNumber(mAddress + 104, 1, 1, 99, value);
		}

		public uint HPPlus
		{
			get => SaveData.Instance().ReadNumber(mAddress + 228, 1);
			set => SaveData.Instance().WriteNumber(mAddress + 228, 1, value);
		}

		public uint AttackPlus
		{
			get => SaveData.Instance().ReadNumber(mAddress + 229, 1);
			set => SaveData.Instance().WriteNumber(mAddress + 229, 1, value);
		}

		public uint DefensePlus
		{
			get => SaveData.Instance().ReadNumber(mAddress + 230, 1);
			set => SaveData.Instance().WriteNumber(mAddress + 230, 1, value);
		}
	}
}

File: monster.py
from save_data import SaveData
from util import Util

class Monster:
    def __init__(self, address):
        self.address = address

    @property
    def name(self):
        return SaveData.instance().read_text(self.address, 30)
    
    @name.setter
    def name(self, value):
        SaveData.instance().write_text(self.address, 30, value)

    @property
    def id(self):
        return SaveData.instance().read_number(self.address + 60, 2)

    @id.setter
    def id(self, value):
        SaveData.instance().write_number(self.address + 60, 2, value)

    @property
    def lv(self):
        return SaveData.instance().read_number(self.address + 104, 1)

    @lv.setter
    def lv(self, value):
        Util.write_number(self.address + 104, 1, 1, 99, value)

    @property
    def hp_plus(self):
        return SaveData.instance().read_number(self.address + 228, 1)

    @hp_plus.setter
    def hp_plus(self, value):
        SaveData.instance().write_number(self.address + 228, 1, value)

    @property
    def attack_plus(self):
        return SaveData.instance().read_number(self.address + 229, 1)

    @attack_plus.setter
    def attack_plus(self, value):
        SaveData.instance().write_number(self.address + 229, 1, value)

    @property
    def defense_plus(self):
        return SaveData.instance().read_number(self.address + 230, 1)

    @defense_plus.setter
    def defense_plus(self, value):
        SaveData.instance().write_number(self.address + 230, 1, value)
